src/frontend/src/components/Header/index.tsx

```diff
@@ -30,12 +32,15 @@ class Header extends React.Component<Props> {
             >
               Stats
             </button>
-            <button
-              onClick={() => history.push('/admin')}
-              className="link mid-gray hover-blue f6 f5-ns dib pa3"
-            >
-              Admin
-            </button>
+            {/* Conditionally render the admin button based on isAdmin from the props */}
```

Comment:
In general I prefer locally scoped comments like this one only if the code does something that is surprising giving the context or unusually unreadable.